test/suite/test_prepare_hs04.py

```diff
@@ -0,0 +1,217 @@
+        # Search keys with timestamp 5, ignore_prepare=false and expect the cursor search to return 0 (key found)
+        self.search_keys_timestamp_and_ignore(ds, txn_config, 0)
+
+        txn_config = 'read_timestamp=' + timestamp_str(20) + ',ignore_prepare=true'
+        # Search keys with timestamp 20, ignore_prepare=true, expect the cursor to return wiredtiger.WT_NOTFOUND
+        self.search_keys_timestamp_and_ignore(ds, txn_config, wiredtiger.WT_NOTFOUND)
+
+        prepare_conflict_msg = '/conflict with a prepared update/'
+        txn_config = 'read_timestamp=' + timestamp_str(20) + ',ignore_prepare=false'
+        # Search keys with timestamp 20, ignore_prepare=false and expect the cursor the cursor search to return prepare conflict message
+        self.search_keys_timestamp_and_ignore(ds, txn_config, prepare_conflict_msg, True)
+
+        # Commit the prepared_transactions with timestamp 30.
+        for j in range (0, self.nsessions):
+            sessions[j].commit_transaction(
+                'commit_timestamp=' + timestamp_str(30) + ',durable_timestamp=' + timestamp_str(30))
+
+        #self.session.commit_transaction()
+        self.session.checkpoint()
+
+        # Simulate a crash by copying to a new directory(RESTART).
+        copy_wiredtiger_home(".", "RESTART")
+
+        # Open the new directory.
+        self.conn = self.setUpConnectionOpen("RESTART")
+        self.session = self.setUpSessionOpen(self.conn)
+
+        # After simulating a crash, search for the keys inserted.
+
+        txn_config = 'read_timestamp=' + timestamp_str(5) + ',ignore_prepare=false'
+        # Search keys with timestamp 5, ignore_prepare=false and expect the cursor search to return 0.
+        self.search_keys_timestamp_and_ignore(ds, txn_config, 0)
+
+        txn_config = 'read_timestamp=' + timestamp_str(20) + ',ignore_prepare=true'
+        # Search keys with timestamp 20, ignore_prepare=true and expect the cursor search to return WT_NOTFOUND.
+        self.search_keys_timestamp_and_ignore(ds, txn_config, wiredtiger.WT_NOTFOUND)
+
+        txn_config = 'read_timestamp=' + timestamp_str(20) + ',ignore_prepare=false'
+        # Search keys with timestamp 20, ignore_prepare=false and expect the cursor search to return WT_NOTFOUND.
+        self.search_keys_timestamp_and_ignore(ds, txn_config, wiredtiger.WT_NOTFOUND)
+
+        txn_config = 'read_timestamp=' + timestamp_str(30) + ',ignore_prepare=true'
+        # Search keys with timestamp 30, ignore_prepare=true and expect the cursor search to return 0.
+        self.search_keys_timestamp_and_ignore(ds, txn_config, 0)
+
+        txn_config = 'read_timestamp=' + timestamp_str(30) + ',ignore_prepare=false'
+        # Search keys with timestamp 30, ignore_prepare=false and expect the cursor search to return 0.
+        self.search_keys_timestamp_and_ignore(ds, txn_config, 0)
+
+        # Close all cursors and sessions, this will cause prepared updates to be
```

Comment:
Can we simulate the crash recovery for the rollback case also?

Also, I don't understand that once the transactions are committed, how the test is again rolling back them. Definitely there is something it is doing which I don't understand, can you explain it here.

Reply:
@ddanderson and @kommiharibabu: I agree to your concern here to close the cursors after restarting. I have updated the code and used `make_scenarios` for two different flows, one to `commit_transaction` and another to `rollback transactions(by simulating crash)`.  Please let me know if that makes sense to you.